pkg/stub/handler.go

```diff
@@ -854,14 +868,30 @@ func (h *Handler) setSampleManagedLabelToFalse(kind, name string) error {
 	return nil
 }
 
-func (h *Handler) createSamples(cfg *v1.Config, updateIfPresent, registryChanged bool, unskippedStreams, unskippedTemplates map[string]bool) error {
+// abortForDelete is used in various portions of the control flow to abort upsert or watch processing
+// if a deletion has been initiated and we want to process the config object's finalizer
+func (h *Handler) abortForDelete(cfg *v1.Config) bool {
+	// reminder refetchCfgMinimizeConflicts accesses the lister/watch cache and does not perform API calls
+	cfg = h.refetchCfgMinimizeConflicts(cfg)
+	if cfg.DeletionTimestamp != nil {
+		return true
+	}
+	return false
+}
+
+// rc:  error - any api errors during upserts
+// rc:  bool - if we abort because we detected a delete
+func (h *Handler) createSamples(cfg *v1.Config, updateIfPresent, registryChanged bool, unskippedStreams, unskippedTemplates map[string]bool) (error, bool) {
```

Comment:
nit - golang convention is return error last
```suggestion
func (h *Handler) createSamples(cfg *v1.Config, updateIfPresent, registryChanged bool, unskippedStreams, unskippedTemplates map[string]bool) (bool, error) {
```

Reply:
param order switched @adambkaplan - thanks !!